vendor/github.com/openshift/cluster-api/pkg/apis/machine/v1beta1/machine_types.go

```diff
@@ -39,6 +39,12 @@ const (
 // Machine is the Schema for the machines API
 // +k8s:openapi-gen=true
```

Comment:
This will be overwritten as a soon as we revendor and we don't want to put it in the source repo as it'll become obsolete as soon as we add a new provider

Reply:
We are not solving the right problem. It's wrong to put anything from `spec.providerSpec`/`.status.providerStatus` under `additionalPrinterColumns.[*].JSONPath` at the first place. Plus, the CRD specification has no conditional `if .field == <value> then print this` statement.

I am against dynamic reconciling of machine CRDs based on a config.

Instead, we SHOULD create a new field in the machine type specification that will get populated by a machine controller in a generic way and accessed by `additionalPrinterColumns.[*].JSONPath`. E.g.

```
newfieldname
aws:instanceID=<instanceID>:instanceState=<instanceState>:instanceType=<instanceType>:region=<region>:zone=<zone>
```

The column width can just get pretty long though.

Reply:
I agree with @ingvagabund here. Things like printer columns should not be reaching into raw-extension fields. We should define generic status fields on the CRD that providers can optionally set and use those.

I also really don't think we should go down the route of modifying our CRD definitions based on provider. Even just to add things like printer columns. I think that is a recipe for disaster when it comes to migrations and fixing issues with providers.

Reply:
yeh so I agree on https://github.com/openshift/machine-api-operator/pull/257#discussion_r273399114 I just don't see how is that related to merge that commit into the vendored folder.
We'll be introducing a new cloud provider soon. I think is more important solve the underlying problem you are stating than having the full automated command